test/built-ins/RegExp/prototype/source/this-val-invalid-obj.js

```diff
@@ -25,3 +27,7 @@ assert.throws(TypeError, function() {
 assert.throws(TypeError, function() {
   get.call(arguments);
 }, 'arguments object');
+
+assert.throws(TypeError, function() {
+  get.call(otherRealmRegExpProto);
+}, 'cross-realm RegExp.prototype');
```

Comment:
Should we also assert that `otherRealmRegExpProto.get.call(RegExp.prototype)` throws, just to be thorough?

Reply:
I don't think so... none of the other cross-realm testing seems to do so, and a failure would be more an indictment of realm construction than of an issue with the `RegExp.prototype.source` getter anyway.

Reply:
Fair enough; I've updated the test and done the same for other special `RegExp.prototype` getters.